From a15d667cedb4ce2db580eb89ed8ad7d9db3e6182 Mon Sep 17 00:00:00 2001
From: Henry Wang <Henry.Wang@arm.com>
Date: Mon, 25 Jul 2022 01:35:58 +0000
Subject: [PATCH] xen/drivers,arm: Move macros FIELD_PREP and FIELD_GET
 to io.h

Before this change, macros `FIELD_PREP` and `FIELD_GET` is private
in `xen/drivers/passthrough/arm/smmu-v3.c`, however, these two
macros are useful when doing MMIO register read/write. To avoid
reinventing the wheel across different drivers, this commit moves
these two macros to the Arm io.h so that other headers can include
these macros easily.

No functional changes intended.

Signed-off-by: Henry Wang <Henry.Wang@arm.com>
Upstream-Status: Pending
Issue-Id: SCM-5049
Change-Id: If8c70e430cca285e6fd3ec1fcfa55f84af620db0
---
 xen/drivers/passthrough/arm/smmu-v3.c | 6 ------
 xen/include/asm-arm/io.h              | 6 ++++++
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/xen/drivers/passthrough/arm/smmu-v3.c b/xen/drivers/passthrough/arm/smmu-v3.c
index d115df7320..c8dfecbb55 100644
--- a/xen/drivers/passthrough/arm/smmu-v3.c
+++ b/xen/drivers/passthrough/arm/smmu-v3.c
@@ -157,12 +157,6 @@ typedef unsigned int		gfp_t;
 #define readl_relaxed_poll_timeout(addr, val, cond, delay_us, timeout_us)	\
 	readx_poll_timeout(readl_relaxed, addr, val, cond, delay_us, timeout_us)
 
-#define FIELD_PREP(_mask, _val)			\
-	(((typeof(_mask))(_val) << (ffs64(_mask) - 1)) & (_mask))
-
-#define FIELD_GET(_mask, _reg)			\
-	((typeof(_mask))(((_reg) & (_mask)) >> (ffs64(_mask) - 1)))
-
 /*
  * Helpers for DMA allocation. Just the function name is reused for
  * porting code, these allocation are not managed allocations
diff --git a/xen/include/asm-arm/io.h b/xen/include/asm-arm/io.h
index e426804424..f0018cfe2a 100644
--- a/xen/include/asm-arm/io.h
+++ b/xen/include/asm-arm/io.h
@@ -9,6 +9,12 @@
 # error "unknown ARM variant"
 #endif
 
+#define FIELD_PREP(_mask, _val)         \
+    (((typeof(_mask))(_val) << (ffs64(_mask) - 1)) & (_mask))
+
+#define FIELD_GET(_mask, _reg)          \
+    ((typeof(_mask))(((_reg) & (_mask)) >> (ffs64(_mask) - 1)))
+
 #endif
 /*
  * Local variables:
-- 
2.25.1

